/Reload  causes sql_export to crash

dear dev:

**Describe the bug**

/Reload causes sql_export to crash

**To Reproduce**
After updating to 0.22.2, I ran /reload multiple times, and the sql_exporter process crashed. Here are the logs:
```
goroutine 85305 [running]:
internal/sync.(*Mutex).Lock(...)
	/usr/local/go/src/internal/sync/mutex.go:63
sync.(*Mutex).Lock(...)
	/usr/local/go/src/sync/mutex.go:46
database/sql.(*DB).conn(0x0, {0x38b8b80, 0xc0028c2fc0}, 0x1)
	/usr/local/go/src/database/sql/sql.go:1317 +0x55
database/sql.(*DB).prepare(0x0, {0x38b8b80, 0xc0028c2fc0}, {0xc000383e60, 0x101}, 0x0?)
	/usr/local/go/src/database/sql/sql.go:1625 +0x39
database/sql.(*DB).PrepareContext.func1(0x0?)
	/usr/local/go/src/database/sql/sql.go:1599 +0x45
database/sql.(*DB).retry(0x0?, 0xc000a1ac78)
	/usr/local/go/src/database/sql/sql.go:1576 +0x42
database/sql.(*DB).PrepareContext(0x38b8b80?, {0x38b8b80?, 0xc0028c2fc0?}, {0xc000383e60?, 0x41c9b4?})
	/usr/local/go/src/database/sql/sql.go:1598 +0x7c
github.com/burningalchemist/sql_exporter.(*Query).run(0xc0017d9180, {0x38b8b80, 0xc0028c2fc0}, 0x0)
	/app/query.go:145 +0x12d
github.com/burningalchemist/sql_exporter.(*Query).Collect(0xc0017d9180, {0x38b8b80, 0xc0028c2fc0}, 0x0, 0xc0028c3500)
	/app/query.go:90 +0xf3
github.com/burningalchemist/sql_exporter.(*collector).Collect.func1(0xc00564acd0?)
	/app/collector.go:82 +0x53
created by github.com/burningalchemist/sql_exporter.(*collector).Collect in goroutine 85303
	/app/collector.go:80 +0x72
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x7c49b5]
```


I downgraded to an older version and the problem is gone. Could this be caused by the previously fixed issue #927?


THK U

--- a/query.go
+++ b/query.go
@@ -125,6 +125,10 @@ func (q *Query) Collect(ctx context.Context, conn *sql.DB, ch chan<- Metric) {
 
 // run executes the query on the provided database, in the provided context.
 func (q *Query) run(ctx context.Context, conn *sql.DB) (*sql.Rows, errors.WithContext) {
+	if conn == nil {
+		return nil, errors.Errorf(q.logContext, "nil database connection")
+	}
+
 	if slog.Default().Enabled(ctx, slog.LevelDebug) {
 		start := time.Now()
 		defer func() {

--- a/target.go
+++ b/target.go
@@ -136,12 +136,13 @@ func (t *target) Collect(ctx context.Context, ch chan<- Metric) {
 	var wg sync.WaitGroup
 	// Don't bother with the collectors if target is down.
 	if targetUp {
+		conn := t.conn
 		wg.Add(len(t.collectors))
 		for _, c := range t.collectors {
 			// If using a single DB connection, collectors will likely run sequentially anyway. But we might have more.
 			go func(collector Collector) {
 				defer wg.Done()
-				collector.Collect(ctx, t.conn, ch)
+				collector.Collect(ctx, conn, ch)
 			}(c)
 		}
 	}

